syntax = "proto3";

option java_package = "ca.eandb.jmist.proto";
option java_outer_classname = "SceneProtos";

import "camera.proto";
import "light.proto";
import "mesh.proto";

message Object {
  enum Type {
    MESH = 0;
  }
  Type type = 1;
  repeated double world_to_local = 2;

  oneof parameters {
    Mesh mesh_object = 3;
  }
}

message Scene {
  Camera camera = 1;
  repeated Light lights = 2;
  repeated Object objects = 3;
}
